packages/components/src/AtlantisThemeContext/AtlantisThemeContext.tsx

```diff
@@ -0,0 +1,129 @@
+  PropsWithChildren,
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import merge from "lodash/merge";
+import {
+  AtlantisThemeContextProviderProps,
+  AtlantisThemeContextValue,
+  THEME_CHANGE_EVENT,
+  Theme,
+} from "./types";
+import { useThemeContextEventQueue } from "./useThemeContextEventQueue";
+import styles from "./AtlantisThemeContext.css";
+
+export const atlantisThemeContextDefaultValues: AtlantisThemeContextValue = {
+  theme: "light",
+  tokens: tokens,
+};
+
+const actualDarkTokens = merge({}, tokens, darkTokens);
+
+const AtlantisThemeContext = createContext(atlantisThemeContextDefaultValues);
+
+export function AtlantisThemeContextProvider({
+  children,
+  dangerouslyForceThemeForProvider,
+}: AtlantisThemeContextProviderProps) {
+  if (dangerouslyForceThemeForProvider) {
+    return (
+      <InternalStaticThemeProvider
+        dangerouslyForceThemeForProvider={dangerouslyForceThemeForProvider}
+      >
+        {children}
+      </InternalStaticThemeProvider>
+    );
+  }
+
+  return (
+    <InternalDynamicThemeProvider>{children}</InternalDynamicThemeProvider>
+  );
+}
+
+function InternalDynamicThemeProvider({ children }: PropsWithChildren) {
+  const initialTheme: Theme =
+    (globalThis.document.documentElement.dataset.theme as Theme) ?? "light";
+
+  const [internalTheme, setInternalTheme] = useState<Theme>(initialTheme);
+  const currentTokens = initialTheme === "dark" ? actualDarkTokens : tokens;
+
+  const {
+    isEmpty,
+    dequeueThemeChange,
+    themeChangeQueue,
+    handleThemeChangeEvent,
+  } = useThemeContextEventQueue();
+
+  useEffect(() => {
+    if (!globalThis.window) return;
+    globalThis.window.addEventListener(
+      THEME_CHANGE_EVENT,
+      handleThemeChangeEvent,
+    );
+
+    return () => {
+      if (!globalThis.window) return;
+      globalThis.window.removeEventListener(
+        THEME_CHANGE_EVENT,
+        handleThemeChangeEvent,
+      );
+    };
+  }, [handleThemeChangeEvent]);
+
+  useEffect(() => {
+    if (isEmpty || !globalThis.document) {
+      return;
+    }
+
+    const newTheme = dequeueThemeChange();
+    setInternalTheme(newTheme);
+  }, [themeChangeQueue, dequeueThemeChange]);
+
+  return (
+    <AtlantisThemeContext.Provider
+      value={{
+        theme: internalTheme,
+        tokens: currentTokens,
+      }}
+    >
+      {children}
+    </AtlantisThemeContext.Provider>
+  );
+}
+
+function InternalStaticThemeProvider({
+  dangerouslyForceThemeForProvider,
+  children,
+}: Required<
+  Pick<
+    AtlantisThemeContextProviderProps,
+    "dangerouslyForceThemeForProvider" | "children"
+  >
+>) {
+  const currentTokens =
+    dangerouslyForceThemeForProvider === "dark" ? actualDarkTokens : tokens;
+
+  return (
+    <AtlantisThemeContext.Provider
+      value={{
+        theme: dangerouslyForceThemeForProvider,
+        tokens: currentTokens,
+      }}
+    >
+      <div
+        data-theme={dangerouslyForceThemeForProvider}
```

Comment:
I kinda disagree on the easiness to grep atlantis. I feel like it adds yet another `data` to remember. Especially if we move away from `data-theme` (for whatever reason) it would be another string to grep to make sure you don't miss something.

I am pretty agnostic though

Reply:
I also think it makes it easier to find all elements overriding a theme in the inspector. You can search something like `*[data-theme="light"], *[data-theme="dark"]` in the inspector to find all elements controlling the theme